dql/models.py

```diff
@@ -105,6 +105,16 @@ def __str__(self):
         else:
             return "QueryIndex(%r, %s, %s)" % (self.name, self.hash_key, self.range_key)
 
+    @property
+    def primary_key_attributes(self):
+        """Get the names of the primary key attributes as a tuple"""
+        if self.hash_key is None:
+            raise ValueError("Missing hash key")
+        if self.range_key is None:
+            return (self.hash_key,)
+        else:
+            return (self.hash_key, self.range_key)
```

Comment:
Required to make column ordering possible. Where main columns are in the front of table.